modules/tournament/src/main/Tournament.scala

```diff
@@ -102,27 +111,135 @@ case class Tournament(
 
   def pairingsClosed = secondsToFinish < pairingsClosedSeconds
 
+  //start at 0 as it's actaully an index for medley variants in front end
   def medleyRound: Option[Int] =
-    if (isStarted) medleyMinutes.map(_ * 60).map((nowSeconds - startsAt.getSeconds).toInt / _)
-    else None
+    if (isStarted) {
+      medleyIntervalSeconds
+        .map(
+          _.map { var s = 0; t => { s += t; s } }
+            .filter(_ <= ((nowSeconds - startsAt.getSeconds).toInt))
+            .size
+        )
+    } else None
+
+  def finalMedleyVariant: Boolean = medleyRound.getOrElse(-99) == medleyNumIntervals.getOrElse(0) - 1
```

Comment:
I personally hate a random default to -99 to prove its false. I think this is trying to say if we don't have a `medleyRound` this needs to be false.
Can it not be rewritten as:
`medleyRound.nonEmpty && medleyRound == medleyNumIntervals.map(_ - 1)`?

Reply:
Yes me too, this is why I shouldn't code while ill :<